scripts/_common.sh

```
@@ -0,0 +1,680 @@
+function apt_repository() {
+  local repository="$1"
+  local source_list="${2-}"
+  local repository_lookup="$repository"
+  if [[ "$repository" == ppa:* ]]; then
+    repository_lookup="${repository##*:}"
+    [[ -n "$source_list" ]] && warn "ppa repository ignores custom source list: $source_list"
+    source_list=
+  fi
+
+  # check if already added
+  local repositories=$(sudo cat /etc/apt/sources.list /etc/apt/sources.list.d/* 2>/dev/null | grep ^[^#])
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)